app-drawer/app-drawer.html

```diff
@@ -552,12 +550,11 @@
 
         if (oldState !== this._drawerState) {
           if (this._drawerState === this._DRAWER_STATE.OPENED) {
+            this._setKeyboardFocusTrap();
             document.addEventListener('keydown', this._boundEscKeydownHandler);
```

Comment:
The 2 listeners are installed on different targets (document and this). The document listener is removed in detached, and the element listener will be GC'd